packages/assets-controllers/src/Standards/ERC20Standard.ts

```diff
@@ -39,24 +32,16 @@ export class ERC20Standard {
    * @returns Promise resolving to the 'decimals'.
    */
   async getTokenDecimals(address: string): Promise<string> {
-    const contract = this.web3.eth.contract(abiERC20).at(address);
-    return new Promise<string>((resolve, reject) => {
-      contract.decimals((error: Error, result: BN | string) => {
-        /* istanbul ignore if */
-        if (error) {
-          reject(error);
-          return;
-        }
-
-        const resultString = result.toString();
-        // We treat empty string or 0 as invalid
-        if (resultString.length > 0 && resultString !== '0') {
-          resolve(resultString);
-        }
+    // Signature for calling `decimals()`
+    const payload = { to: address, data: '0x313ce567' };
```

Comment:
Just curious; why you decided to use `call` here over creating a contract and calling `decimals`? I don't object, but this seems less readable than using the Contract API.

Reply:
We can, but it'll change behaviour of the function.

```
const contract = new Contract(address, abiERC20, this.provider);
const decimals = await contract.decimals();
return decimals.toString();
```

This throws `call revert exception` for our test cases for empty responses. This only changes the error message, but we could rethrow as well, not sure what is preferred.